tests/integration/test_api.py

```diff
@@ -383,6 +389,39 @@ def test_annotation_event(client, monkeypatch, httpserver):
     assert result.status_code == 200
 
 
-def test_display_insight_prediction_for_products(client, mocker):
+def test_insight_prediction(client, mocker):
     result = client.simulate_get("/api/v1/insights/predictions")
     assert result.status_code == 200
+    assert result.json == {"count": 0, "predictions": [], "status": "no_predictions"}
+
+    PredictionFactory(barcode="123")
+    result = client.simulate_get("/api/v1/insights/predictions")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 1
+    assert data["status"] == "found"
+    predictions_data = data["predictions"]
+    assert predictions_data[0]["barcode"] == "123"
+    assert predictions_data[0]["type"] == "category"
+    assert predictions_data[0]["value_tag"] == "en:seeds"
+
+    PredictionFactory(value_tag="en:seeds")
```

Comment:
you should put different values for different attributes here to differentiate and see it works differently.
For example 
```
PredictionFactory(value_tag="en:beers", data={"sample": 1}, type="brand")
```